### PR TITLE
ADD: Gas limit option

### PR DESCRIPTION
**Pull request by PatrickRL**
added gas limit option to validator and execution clients

### CODE REVIEW

File: launcher/src/backend/TekuGasLimitConfig.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  async createGasConfigFile(gasLimit, feeRecipient, configPath){
+    await this.nodeConnection.sshService.exec(`cat > ${configPath}/gas_config.json`);
+    await this.nodeConnection.sshService.exec(`echo '{\n\t"default_config":\n\t{\n\t\t"fee_recipient":"${feeRecipient}",\n\t\t"builder":\n\t\t{\n\t\t\t"enabled":true,\n\t\t\t"gas_limit":"${gasLimit}"\n\t\t}\n\t}\n}' > ${configPath}/gas_config.json`);
+  }
+
+  async removeGasConfigFile(configPath){
+    await this.nodeConnection.sshService.exec(`rm -f ${configPath}/gas_config.json`);
+  }
+
+  async readGasConfigFile(configPath){
```
Comment (gbayasgalan) [inline launcher/src/backend/TekuGasLimitConfig.js:15]: imho would be useful to improve error handling: -> e.g: If the gas_limit is not found or the config file doesn't exist

File: launcher/src/backend/TekuGasLimitConfig.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+export class TekuGasLimitConfig{
+  constructor(nodeConnection) {
+    this.nodeConnection = nodeConnection;
+  }
+
+  async createGasConfigFile(gasLimit, feeRecipient, configPath){
+    await this.nodeConnection.sshService.exec(`cat > ${configPath}/gas_config.json`);
```
Comment (gbayasgalan) [inline launcher/src/backend/TekuGasLimitConfig.js:7]: error handling: if config file doesn't exist

File: launcher/src/backend/TekuGasLimitConfig.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    await this.nodeConnection.sshService.exec(`echo '{\n\t"default_config":\n\t{\n\t\t"fee_recipient":"${feeRecipient}",\n\t\t"builder":\n\t\t{\n\t\t\t"enabled":true,\n\t\t\t"gas_limit":"${gasLimit}"\n\t\t}\n\t}\n}' > ${configPath}/gas_config.json`);
```
Comment (gbayasgalan) [inline launcher/src/backend/TekuGasLimitConfig.js:8]: how about doing more readable like creating json format ---> "const configContent = `{
      "default_config": {
        "fee_recipient": "${feeRecipient}",
        "builder": {
          "enabled": true,
          "gas_limit": "${gasLimit}"
        }
      }
    }`;

    await this.nodeConnection.sshService.exec(`echo '${configContent}' > ${configPath}/gas_config.json`);"
    for better readability also for maintaining

File: launcher/src/backend/TekuGasLimitConfig.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  }
+
+  async removeGasConfigFile(configPath){
+    await this.nodeConnection.sshService.exec(`rm -f ${configPath}/gas_config.json`);
```
Comment (gbayasgalan) [inline launcher/src/backend/TekuGasLimitConfig.js:12]: error handling to check whether file exists or not

**Comment by NeoPlays** [comment]
@PatrickRL I'd start with the stuff @gbayasgalan mentioned. :)

File: launcher/src/backend/TekuGasLimitConfig.js (hunk 3 above)
Comment (NeoPlays) [reply launcher/src/backend/TekuGasLimitConfig.js:8]: we should have quite some functions you can use to escape everything for the shell. see [escapeStringForShell‎](https://github.com/stereum-dev/ethereum-node/blob/main/launcher/src/backend/StringUtils.js#L6)

**Review by NeoPlays — APPROVED** [review]
lgtm

**Pull request merged**